### PR TITLE
tree-sitter: vendor `emscripten`

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
`tree-sitter` requires a specific version of `emscripten` to build, so we are going to vendor it rather than maintain a one-off formula.

Closes https://github.com/Homebrew/homebrew-core/pull/158948

### CODE REVIEW

File: Formula/t/tree-sitter.rb [hunk 1]
```diff
@@ -20,11 +21,30 @@ class TreeSitter < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "d413c47e7b8641aae812d19b39c3f927088e450eabde992f3d7cc32de7d29aa9"
   end
 
-  depends_on "emscripten" => [:build, :test]
   depends_on "node" => [:build, :test]
   depends_on "rust" => :build
+  depends_on "gdbm"
```
Comment (Bo98) [inline Formula/t/tree-sitter.rb:26]: Where does this come from? Installing prebuilts?

Comment (p-linnane) [reply Formula/t/tree-sitter.rb:26]: I believe so. The previous run reported `gdbm` as a broken dependency on Intel, so adding it to see what happens on this run.

Comment (p-linnane) [reply Formula/t/tree-sitter.rb:26]: There's also an audit failure on arm64 with an `emscripten` component that is built only for x86_64, so trying to iterate through some things here.

Comment (Bo98) [reply Formula/t/tree-sitter.rb:26]: Not entirely liking the idea of just converting this to just install prebuilts but acknowledged.

Comment (p-linnane) [reply Formula/t/tree-sitter.rb:26]: I don't like it either. Just trying to find a way to get this to build again that won't blow up when `emscripten` is upgraded. @MikeMcQuaid is on the same page with that distaste.

**Comment by cho-m** [comment]
My opinion is still the same from first PR to just pare down parts that don't meet our requirements (essentially feature deprecation/removal rather than formula deprecation).

From quick glance, I don't see any repository (Alpine, Arch, Debian, Fedora, ...) distributing wasm libs.

From Arch, it looks like `emacs`, `neovim`, and `rizin` only need C libs. No idea on `semgrep` but given we have no `emscripten` dependency then I would assume it isn't using wasm feature.

**Comment by Bo98** [comment]
I'm mainly a bit concerned that the decision to switch to install prebuilts here is the start of a bad habit of gravitating towards prebuilts whenever an awkward source build breaks, particularly given this isn't even a critical feature of tree-sitter and seems to be just something that enables using the web UI without an internet connection.

We already required the user to install emscripten themselves if they want to use `tree-sitter build-wasm` so that part isn't really a concern (except maybe our ability to test it but we don't have to test that part) and is a usability issue that's not our doing.

**Comment by MikeMcQuaid** [comment]
> I'm mainly a bit concerned that the decision to switch to install prebuilts here

My understanding was these were just build-time dependencies that would not be installed if used as a `resource`. If not: I agree, we should not do this.


> From quick glance, I don't see any repository (Alpine, Arch, Debian, Fedora, ...) distributing wasm libs.
>
> particularly given this isn't even a critical feature of tree-sitter and seems to be just something that enables using the web UI without an internet connection

Agreed. Let's remove this functionality from `tree-sitter` instead.

@p-linnane sorry for all the back-and-forth here!

**Comment by p-linnane** [comment]
Opened https://github.com/Homebrew/homebrew-core/pull/159082 to hopefully finally put this ordeal to rest! 😅 Thanks for all the input everyone.

**Pull request closed without merging**